File: 알고리즘 정리/스택,큐/문제풀이/기능개발.cpp
#include <iostream>
////////////////////////////////////////////////////////////////////////
#include <string>
#include <vector>
using namespace std;

vector<int> solution(vector<int> progresses, vector<int> speeds) {
	vector<int> answer;
	while (!progresses.empty())
	{
		int d = (100 - progresses[0]) / speeds[0];
		if ((100 - progresses[0]) % speeds[0] != 0)
			++d;
		progresses.erase(progresses.begin());
		speeds.erase(speeds.begin());
		int cnt = 1;
		while (!progresses.empty() && progresses[0] + d * speeds[0] >= 100)
		{
			++cnt;
			progresses.erase(progresses.begin());
			speeds.erase(speeds.begin());
		}
		answer.push_back(cnt);
	}
	return answer;
}
////////////////////////////////////////////////////////////////////////

int main()
{
	cin.sync_with_stdio(0);
	cin.tie(0); cout.tie(0);
	vector<int> progresses = { 93,30,55 };
	vector<int> speeds = { 1,30,5 };
	vector<int> result = solution(progresses, speeds);
	for (int i = 0; i < result.size(); i++)
		cout << result[i] << ' ';
	cout << '\n';
	return 0;
}

File: 알고리즘 정리/스택,큐/문제풀이/기능개발.py
####################################################################
from math import ceil

def solution(progresses, speeds):
    answer = []
    while progresses:
        d=ceil((100-progresses[0])/speeds[0]) #처음 위치가 배포되는데 걸리는 날의 수
        cnt=1 # 한번 배포에 몇개가 배포되는지 count
        progresses.pop(0)
        speeds.pop(0)
        while len(progresses)!=0 and progresses[0]+d*speeds[0]>=100:
            cnt+=1
            progresses.pop(0)
            speeds.pop(0)
        answer.append(cnt)
    return answer

########################################################################

progresses=[93,30,55]
speeds=[1,30,5]
print(solution(progresses,speeds))